src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -284,19 +300,13 @@ private async Task ListenForMessages(CancellationToken cancellationToken)
             Message messageToDispatch,
             MessageWriter messageWriter)
         {
-            Task handlerToAwait = null;
+            Func<Message, MessageWriter, Task> handlerToAwait = null;
```

Comment:
changed hanlderToAwait to a Func instead of a Task and removed `handlerToAwait = requestHandler(messageToDispatch, messageWriter);` so we don't have to wait for the potential slow sync code in the handler. The invocation of hanlderToAwait happens later when we awaits.